packages/text-input/src/TextInput.tsx

```diff
@@ -323,12 +325,18 @@ const TextInput: React.ComponentType<
       }
     }
 
-    if (!label && !ariaLabelledby) {
+    if (type !== 'search' && !label && !ariaLabelledby) {
```

Comment:
Nit, but this might be more readable if written like:

```js
if (type === 'search') {
  if (!label && !ariaLabelledby) {
     console.error('...');
  }
} else {
  if (!rest['aria-label']) {
    console.error('...');
  }
}
```